File: ios/Momentum/Models/Coach.swift
import Foundation

struct CoachMessage: Codable, Identifiable { let id: UUID; let date: Date; let text: String; let isUser: Bool }
struct Challenge: Codable, Identifiable { let id: UUID; let title: String; let description: String; let duration: Int; let category: String; let points: Int; let isActive: Bool }
struct JournalEntry: Codable, Identifiable { let id: UUID; let date: Date; let text: String; let moodScore: Int }
struct FocusSession: Codable, Identifiable { let id: UUID; let date: Date; let durationMinutes: Int; let completed: Bool }
struct PersonalizedContent: Codable, Identifiable { let id: UUID; let title: String; let description: String; let type: ContentType; let relevanceScore: Double; let imageURL: String? }
enum ContentType: String, Codable { case article = "Article"; case video = "Video"; case exercise = "Exercise"; case recipe = "Recipe" }

File: android/app/src/main/java/com/momentum/app/models/Coach.kt
package com.momentum.app.models

import java.util.UUID

data class CoachMessage(
    val id: String = UUID.randomUUID().toString(),
    val type: MessageType,
    val title: String,
    val body: String,
    val actionTitle: String? = null,
    val actionType: String? = null,
    val timestamp: Long = System.currentTimeMillis(),
    val priority: Int = 0
)

enum class MessageType {
    MorningBriefing, EveningReview, Alert, Suggestion, Achievement, CheckIn
}

data class Challenge(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val description: String,
    val durationDays: Int,
    val category: MicroActionCategory,
    val reward: String,
    val progress: Int = 0,
    val completed: Boolean = false
)

data class JournalEntry(
    val id: String = UUID.randomUUID().toString(),
    val content: String,
    val mood: Int = 5,
    val energyBefore: EnergyLevel = EnergyLevel.Moderate,
    val energyAfter: EnergyLevel = EnergyLevel.Moderate,
    val reflectionQuestions: List<Pair<String, String>> = emptyList(),
    val timestamp: Long = System.currentTimeMillis()
)

data class FocusSession(
    val id: String = UUID.randomUUID().toString(),
    val durationMinutes: Int = 25,
    val completedMinutes: Int = 0,
    val interrupted: Boolean = false,
    val focusScoreAfter: Int? = null,
    val timestamp: Long = System.currentTimeMillis()
)

data class PersonalizedContent(
    val id: String = UUID.randomUUID().toString(),
    val type: ContentType,
    val title: String,
    val subtitle: String,
    val body: String,
    val actionLabel: String? = null
)

enum class ContentType {
    Article, Tip, Exercise, Recipe, Meditation, Challenge
}
